lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,183 @@
+                continue
+            result_dir = dirent
+            mrt = MakeResultTb(
+                result_dir,
+                temp_dir,
+                self.context.user,
+                self.context.prefix,
+                self.config,
+                logger,
+            )
 
-    if runs_copied + failures > 0:
-        logger.debug(
-            "successfully moved %s runs, encountered %s failures", runs_copied, failures
-        )
+            try:
+                make_return_val = mrt.check_result_target_dir(result_dir, temp_dir)
+            except FileNotFoundError as e:
+                logger.error(e)
+
+            if make_return_val:
+                failures += 1
+                continue
+            status, result_tb_name = mrt.make_result_tb()
+
+            if status == 0:
+                if os.path.exists(result_tb_name):
+                    assert (
+                        result_tb_name
+                    ), "Logic bomb!  make_result_tb() always returns a tar ball name"
+
+                    crt = CopyResultTb(
+                        self.context.controller, result_tb_name, self.config, logger
+                    )
+                    copy_return_val = crt.check_tb_tbmd5()
+                    if copy_return_val:
+                        failures += 1
+                        continue
+                    ret = crt.copy_result_tb(self.context.token)
+                    if ret:
+                        failures += 1
+                        continue
+            else:
+                logger.error("Error occured while creating Result Tarball")
+                failures += 1
+                continue
+
+            try:
+                # We always remove the constructed tar ball, regardless of success
+                # or failure, since we keep the result directory below on failure.
+                os.remove(result_tb_name)
+                os.remove(f"{result_tb_name}.md5")
+            except OSError:
+                logger.error("rm failed to remove %s and its .md5 file", result_tb_name)
+                failures += 1
+                continue
+
+            try:
+                shutil.rmtree(result_dir)
+            except OSError:
+                logger.error(
+                    "rm failed to remove the %s directory hierarchy", result_dir
+                )
+                failures += 1
+                continue
+
+            runs_copied += 1
+
+        if runs_copied + failures > 0:
+            logger.debug(
+                "successfully moved %s runs, encountered %s failures",
+                runs_copied,
+                failures,
+            )
+            return 1
+
+        return 0
+
+
+@click.command()
+@common_options
+@click.option(
+    "--controller",
+    required=True,
+    envvar="_pbench_full_hostname",
+    prompt=False,
+    help="Controller name",
+)
+@click.option(
+    "--user", default="pbench", prompt=False, help="Pbench user",
+)
```

Comment:
So, here's an interesting conundrum:

- we need the token in order to get access to the server
- apparently we want to record the user in the metadata
- the user is already encoded in the token

What happens if the value of `user` doesn't match the one in the token?!

We have two sources of truth, here, and we're not confirming that they match.  At a minimum, I think we need to validate the match; however, it would be better still if we could avoid having an option on this command for specifying the user.  

Is there some well-defined way to extract the user from the token?  (I.e., can we do this on the client side?...I am unenthusiastic about the idea of calling the server to decode the token...for several reasons.)

Or, can we omit the user from the metadata or do the addition on the server side (where it knows the user)?  Or, perhaps we should record the _token_ in the metadata, instead?  (@dbutenhof , @portante)

Reply:
We should remove `--user` and `--prefix` entirely. They are no longer needed as they were to help organize data before we had the notion of ownership.

Reply:
I was thinking about that earlier, and whether we wanted to retain that sort of layering on top of the real user. I'm happy to agree with you to just drop it.

We do want to work towards allowing arbitrary user metadata, with the ability to search and sort by metadata key values; that should fill any real need for stuff like this.

Reply:
I will just remove it